src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsIpc/IpcTransport.cs

```diff
@@ -48,6 +48,88 @@ internal abstract class IpcEndpoint
         public abstract Task WaitForConnectionAsync(CancellationToken token);
     }
 
+    internal class IpcEndpointConfig
+    {
+        internal enum PortType
```

Comment:
Wasn't sure we wanted to include this in any public API at this point since it's mainly for our internal tools, maybe we should start with internal and then if there is a need, switch to public?